cpp/src/unary/cast_ops.cu

```diff
@@ -194,7 +194,7 @@ std::unique_ptr<column> rescale(column_view input,
       auto const scalar  = make_fixed_point_scalar<T>(0, scale_type{scale}, stream);
       auto output_column = make_column_from_scalar(*scalar, input.size(), stream, mr);
       if (input.nullable()) {
-        auto const null_mask = detail::copy_bitmask(input, stream, mr);
+        auto null_mask = detail::copy_bitmask(input, stream, mr);
         output_column->set_null_mask(std::move(null_mask), input.null_count());
```

Comment:
oof. I thought this woudn't compile. This seems pretty hard to find in general.

Reply:
`std::move` doesn't move anything. It just tries to convert the parameter into `rvalue`. So for this case it failed to convert and the nul mask was just copied.

Reply:
in this case, it called the overloaded function `set_null_mask` with `lvaue` rmm::buffer, and defaulted stream. So inside that `set_null_mask`, it copied the buffer.